cmd/oras/push.go

```diff
@@ -186,9 +185,6 @@ func packManifest(ctx context.Context, store *file.Store, annotations map[string
 	packOpts.ConfigAnnotations = annotations[option.AnnotationConfig]
 	packOpts.ManifestAnnotations = annotations[option.AnnotationManifest]
 
-	if opts.artifactType != "" {
-		packOpts.ConfigMediaType = opts.artifactType
-	}
 	if opts.manifestConfigRef != "" {
 		path, mediatype := parseFileReference(opts.manifestConfigRef, oras.MediaTypeUnknownConfig)
```

Comment:
The `ArtifactType` is defined by the config media type.
```suggestion
		mediaType := oras.MediaTypeUnknownConfig
		if opts.artifactType != "" {
			mediaType = opts.artifactType
		}
		path, mediaType := parseFileReference(opts.manifestConfigRef, mediaType)
```

Reply:
It's the same, since `opts.artifactType` and `opts.manifestConfigRef` are not allowed to be both provided.

Reply:
It is not the same. What if someone do
```
oras push --artifact-type test.artifact --config test.json
```
or
```
oras push --artifact-type test.artifact --config test.json:test.config
```
What's media type of the resulted config? The first one should be `test.artifact` instead of unknown config media type. The second one should remain `test.config` instead of `test.artifact`.